### ISSUE TITLE
[Bug] the return value is ignored when use StreamSink

### ISSUE THREAD

**Issue by mikai233** [fzyzcjy/flutter_rust_bridge#1020]
### Describe the bug

```rust
pub fn new_connection(addr: String, sink: StreamSink<String>) -> anyhow::Result<String> {
    // crate::net::new_connection(addr)
    Ok("".to_string())
}
```
and the the generated dart code like this:
![image](https://user-images.githubusercontent.com/49525397/216091056-700a8dd6-f141-4ee9-856e-03e57380c17f.png)
the return value string is ignored
version of flutter_rust_bridge 1.62.0

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
not relevant
```

**Comment by fzyzcjy** [comment]
Yes, currently it is like that, because the generated Dart code returns a Stream, thus cannot return "a stream + a normal value". However, in your case, what about 

```
pub fn new_connection(addr: String, sink: StreamSink<String>) {
    sink.add("".to_string())
}
```

**Comment by mikai233** [comment]
> 

thanks, actually my code has different type between StreamSink and return value, what i want is start a connection in rust and return a msg sender to dart side send msg to network, and the streamSkink is used to receive msg resp, any good idea about this ?

**Comment by mikai233** [comment]
> Yes, currently it is like that, because the generated Dart code returns a Stream, thus cannot return "a stream + a normal value". However, in your case, what about
> 
> ```
> pub fn new_connection(addr: String, sink: StreamSink<String>) {
>     sink.add("".to_string())
> }
> ```
thanks, actually my code has different type between StreamSink and return value, what i want is start a connection in rust and return a msg sender to dart side send msg to network, and the streamSkink is used to receive msg resp, any good idea about this ?

**Comment by fzyzcjy** [comment]
What about 

```
StreamSink<MyUnionType>

enum MyUnionType {
  A(TheTypeThatYouWantAsStream),
  B(TheTypeYouWantAsReturn),
}

**Comment by mikai233** [comment]
> What about
> 
> ```
> StreamSink<MyUnionType>
> 
> enum MyUnionType {
>   A(TheTypeThatYouWantAsStream),
>   B(TheTypeYouWantAsReturn),
> }
> ```

thanks, it works

**Comment by fzyzcjy** [comment]
Happy to hear that!
